Code/ForceField/MMFF/Params.cpp

```diff
@@ -5091,28 +5214,39 @@ const std::string defaultMMFFDfsb =
     "3	2	4	0.25	0.25\n"
     "4	2	4	0.25	0.25\n";
 
-class MMFFOopCollection *MMFFOopCollection::ds_instance[2] = {NULL, NULL};
+boost::scoped_ptr<MMFFOopCollection> MMFFOopCollection::ds_instance[2];
+#ifdef BOOST_THREAD_PROVIDES_ONCE_CXX11
+boost::once_flag MMFFOopCollection::ds_flag;
+#else
+boost::once_flag MMFFOopCollection::ds_flag = BOOST_ONCE_INIT;
+#endif
 
 extern const std::string defaultMMFFOop;
 extern const std::string defaultMMFFsOop;
 
 MMFFOopCollection *MMFFOopCollection::getMMFFOop(const bool isMMFFs,
                                                  const std::string &mmffOop) {
   unsigned int i = (isMMFFs ? 1 : 0);
-  if (!ds_instance[i]) {
-    ds_instance[i] = new MMFFOopCollection(isMMFFs, mmffOop);
-  } else if (mmffOop != "") {
-    delete ds_instance[i];
-    ds_instance[i] = new MMFFOopCollection(isMMFFs, mmffOop);
+#ifdef RDK_THREADSAFE_SSS
+  boost::call_once(ds_flag, boost::bind(&create, isMMFFs, mmffOop));
+#else
+  static bool created = false;
+  if (!created || !mmffOop.empty()) {
+    created = true;
+    create(isMMFFs, mmffOop);
   }
-  return ds_instance[i];
+#endif
+  return ds_instance[i].get();
```

Comment:
Note, for the above reason this becomes a potential crash. If one thread asks for MMFFs and the other MMFF, then they will race to see who initializes ds_flag first, the 's', or non-'s'. 

The following may be a viable solution (modulo the above parameter re-initialization problem). 

```cpp
boost::once_flag MMFFOopCollection::ds_flag[2] = {BOOST_ONCE_INIT, BOOST_ONCE_INIT};
...

boost::call_once(ds_flag[i], boost::bind(&create, isMMFFs, mmffOop));

```